### DISCUSSION TITLE
write access token in jwt callback with TypeScript

### DISCUSSION THREAD

**Discussion by kyduff** [nextauthjs/next-auth#4735] [Help] [answered]
I'd like to extract the access token from my authentication session to pass in the header of other API calls in an application. The [docs](https://next-auth.js.org/configuration/callbacks#jwt-callback) suggest doing this by writing the access token to an additional field on the `token` object in the `jwt` callback when the user signs in:

```javascript
callbacks: {
  async jwt({ token, account }) {
    // Persist the OAuth access_token to the token right after signin
    if (account) {
      token.accessToken = account.access_token
    }
    return token
  }
}
```

Confusingly the `token` object is a `JWT` type which doesn't have an `accessToken` field, producing an unknown type that breaks any downstream type checks. Unfortunately I can't find a workaround with a union or extending type because the `jwt` callback is expected to produce a `JWT`.

Has anyone else dealt with this problem and found a solution?

**Comment by BetweenTwoTests** [comment, marked as answer]
JWT does not have accessToken field as you said. You would need to use module augmentation as per documentation if you are using typescript. I believe the code you posted works fine for plain JavaScript.

Doc link: https://next-auth.js.org/getting-started/typescript#module-augmentation

Here's what I do to log the callbacks for learning the flow
![image](https://user-images.githubusercontent.com/10789273/174841920-3596b7a5-28a2-404f-8079-5a3b65c883b8.png)

**Reply by kyduff** [reply]
Thanks!

**Comment by LOGOLFGODORD** [comment]
If any of this is about my account accesses, please make sure to contact me
privately Immediately before you do anything ! thanks ! Code of conduct
please!

Thanks,

Justin
ᐧ
